start/common/src/main/java/org/ethereum/beacon/bench/BenchmarkingBeaconChainSpec.java

```diff
@@ -0,0 +1,298 @@
+package org.ethereum.beacon.bench;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import org.ethereum.beacon.consensus.BeaconChainSpec;
+import org.ethereum.beacon.consensus.hasher.ObjectHasher;
+import org.ethereum.beacon.consensus.util.CachingBeaconChainSpec;
+import org.ethereum.beacon.core.BeaconBlock;
+import org.ethereum.beacon.core.BeaconState;
+import org.ethereum.beacon.core.MutableBeaconState;
+import org.ethereum.beacon.core.operations.Attestation;
+import org.ethereum.beacon.core.operations.attestation.AttestationData;
+import org.ethereum.beacon.core.spec.SpecConstants;
+import org.ethereum.beacon.core.state.ShardCommittee;
+import org.ethereum.beacon.core.state.ValidatorRecord;
+import org.ethereum.beacon.core.types.BLSPubkey;
+import org.ethereum.beacon.core.types.BLSSignature;
+import org.ethereum.beacon.core.types.Bitfield;
+import org.ethereum.beacon.core.types.EpochNumber;
+import org.ethereum.beacon.core.types.Gwei;
+import org.ethereum.beacon.core.types.SlotNumber;
+import org.ethereum.beacon.core.types.ValidatorIndex;
+import org.ethereum.beacon.crypto.BLS381.PublicKey;
+import org.ethereum.beacon.util.stats.MeasurementsCollector;
+import org.ethereum.beacon.util.stats.TimeCollector;
+import tech.pegasys.artemis.ethereum.core.Hash32;
+import tech.pegasys.artemis.util.bytes.BytesValue;
+import tech.pegasys.artemis.util.collections.ReadList;
+import tech.pegasys.artemis.util.uint.UInt64;
+
+public class BenchmarkingBeaconChainSpec extends CachingBeaconChainSpec {
```

Comment:
If we have any way to see when we forget to `override` smth. But I don't see any.

Reply:
Basically, I can see only one problem around this. If _some_ method is not overridden by this class while it's been enlisted in one of benchmarking routine, e.g. `BLS` or `HELPERS`. Then this method won't be tracked and counted in the report. Possible solution here is to check whether methods specified in those lists are overridden in `BenchmarkingBeaconChainSpec` but this solution implies usage of reflection. Am not sure that this is a worth thing.